### ISSUE TITLE
Should the color mode expose a variable or event that can be used by third-party components?

### ISSUE THREAD

**Issue by ZhangChengLin** [twbs/bootstrap#39266]
### Prerequisites

- [X] I have [searched](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed feature requests
- [X] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

### Proposal

Should a parameter of the current color mode value be exposed for use by other components?

And whether there should be an active event that can be monitored when the color mode of the page changes.

### Motivation and context

For example, when using other components on the page, third-party components also support bright mode and other mode switching, and can be modified synchronously with the bootstrap color mode status. Should bootstrap provide convenience for this?

**Comment by julien-deramond** [comment]
> Should a parameter of the current color mode value be exposed for use by other components?

Are you talking about the color mode value used for the whole page? Because the color mode can be used contextually within the page too.

> And whether there should be an active event that can be monitored when the color mode of the page changes.

[Our `color-modes.js`](https://github.com/twbs/bootstrap/blob/main/site/static/docs/5.3/assets/js/color-modes.js) is not embedded in the Bootstrap JS file. Whenever the color mode changes for the whole page, I'd say one could add an event in this code to modify synchronously third-party components for instance, or to trigger anything.

Do you have some technical suggestions based on what already exists in Bootstrap?

**Comment by ZhangChengLin** [comment]
> Are you talking about the color mode value used for the whole page? Because the color mode can be used contextually within the page too.

Yes, I mean that when the color mode of the entire page level is modified by the user through buttons, can know that the color mode of the page has been adjusted through event monitoring without refreshing the page.

I understand that color mode can change the color mode locally in the page, but I currently don't know the relevant scenarios where it is needed.


> [Our `color-modes.js`](https://github.com/twbs/bootstrap/blob/main/site/static/docs/5.3/assets/js/color-modes.js?rgh-link-date=2023-10-03T06%3A20%3A08Z) is not embedded in the Bootstrap JS file. Whenever the color mode changes for the whole page, I'd say one could add an event in this code to modify synchronously third-party components for instance, or to trigger anything.

Yes, that's fine if there is.


> Do you have some technical suggestions based on what already exists in Bootstrap?

There is not

**Comment by 2dareis2do** [comment]
You can use css variables for this. e.g. in your css you have something like this:

```
:root, [data-bs-theme=light]{

--bs-blue: #0d6efd;
...

}

We can see that $blue is defined as a scss variable

```
$blue: #0d6efd !default;
```

Then again is scss we have somehting like

```
[data-bs-theme="blue"] {
...
  --bs-body-bg: var(--bs-blue);
```

```

But these do not appear to be used or fully supported when switching from say light to dark mode. What we do have though are examples where it does. e.g. --bs-body-color

```
:root, [data-bs-theme=light] {
--bs-body-color: #212529;
...

}

```

also 

```
@media (prefers-color-scheme:dark)
:root {
color-scheme: dark;
--bs-body-color: #dee2e6;
```

so here you can access like element: var(--bs-blue)

Interesting the only references to this variable I can find are 

```
<body style="--bs-body-color: #333;">
```

and

```
<!DOCTYPE html>

// scss-docs-start custom-color-mode
--
[data-bs-theme="blue"] {
--bs-body-color: var(--bs-white);


```

Whats interesting about this is assigning or overriding one css variable with another in the scss>

Tbh, I am somewhat surprised  that it is not being used like so:

```
body {
  color: var(--bs-body-color)
}
```

and then for dark mode we could do 

```
  @media (prefers-color-scheme: dark) {
    --bs-body-color: var(--bs-white);
  }

```

Also you can change this value in a differrent context.

I think one issue with colours in bootstrap is they use a combination of css and scss variables. Maybe this will be rationalised in boostrap 6. Is there a version of boostrap 6 available yet?

**Comment by mdo** [comment]
`color-scheme` values are available, as is a `data-bs-theme` on the root element if you use that.
